lib/account.js

```diff
@@ -280,12 +280,23 @@ class Account {
             actions: [transaction_1.functionCall(methodName, args, gas || constants_1.DEFAULT_FUNCTION_CALL_GAS, amount)]
         });
     }
-    functionCallV2({ contractId, methodName, args = {}, gas = constants_1.DEFAULT_FUNCTION_CALL_GAS, attachedDeposit, walletMeta, walletCallbackUrl, stringify }) {
+    functionCallV2({ contractId, methodName, args = {}, gas = constants_1.DEFAULT_FUNCTION_CALL_GAS, attachedDeposit, walletMeta, walletCallbackUrl, stringify, jsContract }) {
         this.validateArgs(args);
-        const stringifyArg = stringify === undefined ? transaction_1.stringifyJsonOrBytes : stringify;
+        let functionCallArgs;
+        if (jsContract) {
+            function encodeCall(contract, method, args) {
+                return Buffer.concat([Buffer.from(contract), Buffer.from([0]), Buffer.from(method), Buffer.from([0]), Buffer.from(args)]);
+            }
+            let encodedArgs = encodeCall(contractId, methodName, JSON.stringify(Object.values(args)));
```

Comment:
The arguments need to be passed in like:

`'["arg1_is_string", "arg2_is_string", arg3_is_value, ..., "argn_is_string"]'`

Reply:
I'm a little unclear on what you mean @BenKurrek

Reply:
This actually was changed back so that arguments can be passed in using key value pairs as you have it now. Disregard my previous comment. The way it was before is that you needed to pass in arguments in an array (just the value and not key value pairs) so if I had a function that had a parameter accountId I would pass in '["benji.testnet"]' instead of '{"accountId": "benji.testnet"}' and if it's not a string, you just wouldn't use the quotes.